### PR TITLE
refactor node_test.go

### PR DESCRIPTION
**Pull request by faddat**
- refactor TestNodeNewNodeGenesisHashMismatch
- Refactor tests and add new test cases

This PR should fix several frequent test failures identified by running the tests 40x.

Like with the others, please check this PR harshly, but I am high confidence on this one. 

Closes: #1926
Closes: #1927
Closes: #1928

---

#### PR checklist

- [x] Tests written/updated
- [ ] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [ ] Updated relevant documentation (`docs/` or `spec/`) and code comments

### CODE REVIEW

File: node/node_test.go [hunk 1]
```diff
@@ -170,10 +191,18 @@ func TestPprofServer(t *testing.T) {
...
 
-	resp, err := http.Get("http://" + config.RPC.PprofListenAddress + "/debug/pprof")
-	assert.NoError(t, err)
-	defer resp.Body.Close()
-	assert.Equal(t, 200, resp.StatusCode)
+	// Retry mechanism
+	for i := 0; i < 5; i++ {
+		resp, err := http.Get("http://" + config.RPC.PprofListenAddress + "/debug/pprof")
+		if err == nil {
+			defer resp.Body.Close()
```
Comment (melekes) [inline node/node_test.go:198]:
```suggestion
			errClose := resp.Body.Close()
			require.NoError(t, errClose)
```

**Review by melekes — COMMENTED** [review]
Thanks @faddat ❤️ Unsure how `TestNodeNewNodeGenesisHashMismatch` and `TestNodeDelayedStart` are fixed by changing/adding comments though. Could you please explain?

**Comment by faddat** [comment]
@melekes I think we should run the tests another 100x.

In cases where there's just comments, it is me getting a better grip on the code. 

I'll just grind on all this some so that we can use CI as a source of truth (thats the goal of this exercise)

**Comment by faddat** [comment]
closing in favor of #1954

**Pull request closed without merging**